mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -1116,70 +1132,62 @@ def acknowledge(self):
+        If prefere_translations is true it will lokk in translations
+        directory first. Otherwise in skill data folders and then in
+        translations directory.
+
+        Translation_dir and prefere_translaion can be set under skill
+        in mycroft.conf.
+
         Arguments:
             root_directory (str): root folder to use when loading files.
         """
         root_directory = root_directory or self.root_dir
-        self.init_dialog(root_directory)
-        self.load_vocab_files(root_directory)
-        self.load_regex_files(root_directory)
-
-    def load_vocab_files(self, root_directory):
-        """ Load vocab files found under root_directory.
+        translations_dir = join(self.translations_dir,
+                                self.skill_id,
+                                self.lang)
+        data_folders = []
+        for folder in ['vocab', 'dialog', 'regex', 'locale']:
+            if isdir(join(root_directory, folder, self.lang)):
+                data_folders.append(join(root_directory, folder, self.lang))
+        if isdir(translations_dir) and data_folders == []:
+            data_folders.append(translations_dir)
+        elif data_folders == []:
+            # fallback to en-us or old non transabael resource folders
+            for folder in ['vocab', 'dialog', 'regex', 'locale']:
+                if isdir(join(root_directory, folder, 'en-us')):
+                    data_folders.append(join(root_directory,
+                                             folder,
+                                             self.lang))
+                else:
+                    data_folders.append(join(root_directory, folder))
 
-        Arguments:
-            root_directory (str): root folder to use when loading files
-        """
-        keywords = []
-        vocab_dir = join(root_directory, 'vocab', self.lang)
-        locale_dir = join(root_directory, 'locale', self.lang)
-        if exists(vocab_dir):
-            keywords = load_vocabulary(vocab_dir, self.skill_id)
-        elif exists(locale_dir):
-            keywords = load_vocabulary(locale_dir, self.skill_id)
+        if self.prefere_translations and exists(translations_dir):
+            self.dialog_renderer = load_dialogs(translations_dir)
+            vocabs = load_vocabulary(translations_dir, self.skill_id)
+            regexes = load_regex(translations_dir, self.skill_id)
         else:
-            LOG.debug('No vocab loaded')
-
-        # For each found intent register the default along with any aliases
-        for vocab_type in keywords:
-            for line in keywords[vocab_type]:
-                entity = line[0]
-                aliases = line[1:]
-                self.intent_service.register_adapt_keyword(vocab_type,
-                                                           entity,
-                                                           aliases)
-
-    def load_regex_files(self, root_directory):
```

Comment:
Same goes for this.

Reply:
Great and thanks. I will redo my work and keep the funcions and see if I can figure out what jenkins says....